tests/vppinputoutput.cpp

```diff
@@ -193,17 +193,29 @@ bool VppOutput::getFormat(uint32_t& fourcc, int& width, int& height)
 bool VppOutputFile::init(const char* outputFileName, uint32_t fourcc, int width,
                          int height, const char* /*codecName*/)
 {
+    uint32_t outputFourcc = 0x00;
     if (!outputFileName) {
         ERROR("output file name is null");
         return false;
     }
-    if (!fourcc || !width || !height) {
-        if (!guessFormat(outputFileName, fourcc, width, height)) {
-            ERROR("can't guess format from %s", outputFileName);
+    if (!width || !height) {
+        if (!guessResolution(outputFileName, width, height)) {
+            ERROR("failed to guess file(%s) width and height\n", outputFileName);
             return false;
         }
     }
-    m_fourcc = fourcc;
+
+    outputFourcc = guessFourcc(outputFileName);
```

Comment:
if the fourcc is not 0, we need use the fourcc, if foucc is 0, we can guess fourcc from the file name